### PR TITLE
Change to skip running included tasks when condition not meet and refine unnecessary filter

### PR DESCRIPTION
**Pull request by Tomorrow9**
Some `| lower` and `| bool` are not necessary since the variable is already in the lower case or bool type when defined and set.

### CODE REVIEW

**Comment by Tomorrow9** [comment]
```
VM information:
+----------------------------------------------------------------------------------+
| Name                      | test_vm                                              |
+----------------------------------------------------------------------------------+
| Guest OS Distribution     | RedHat 9.4 x86_64                                    |
+----------------------------------------------------------------------------------+
| IP                        |                                                                |
+----------------------------------------------------------------------------------+
| GUI Installed             | True                                                 |
+----------------------------------------------------------------------------------+
| CloudInit Version         | 23.4                                                 |
+----------------------------------------------------------------------------------+
| Hardware Version          | vmx-14                                               |
+----------------------------------------------------------------------------------+
| VMTools Version           | 12.3.5 (build-22544099)                              |
+----------------------------------------------------------------------------------+
| Config Guest Id           | rhel8_64Guest                                        |
+----------------------------------------------------------------------------------+
| Guest Short Name          |                                                      |
+----------------------------------------------------------------------------------+
| GuestInfo Guest Id        | rhel9_64Guest                                        |
+----------------------------------------------------------------------------------+
| GuestInfo Guest Full Name | Red Hat Enterprise Linux 9 (64-bit)                  |
+----------------------------------------------------------------------------------+
| GuestInfo Guest Family    | linuxGuest                                           |
+----------------------------------------------------------------------------------+
| GuestInfo Detailed Data   | architecture='X86'                                   |
|                           | bitness='64'                                         |
|                           | cpeString='cpe:/o:redhat:enterprise_linux:9::baseos' |
|                           | distroAddlVersion='9.4 (Plow)'                       |
|                           | distroName='Red Hat Enterprise Linux'                |
|                           | distroVersion='9.4'                                  |
|                           | familyName='Linux'                                   |
|                           | kernelVersion='5.14.0-427.13.1.el9_4.x86_64'         |
|                           | prettyName='Red Hat Enterprise Linux 9.4 (Plow)'     |
+----------------------------------------------------------------------------------+


Test Results (Total: 31, Passed: 29, Skipped: 2, Elapsed Time: 02:44:12)
+-------------------------------------------------------------------------+
| ID | Name                                 |   Status        | Exec Time |
+-------------------------------------------------------------------------+
| 01 | deploy_vm_efi_paravirtual_vmxnet3    |   Passed        | 00:31:28  |
| 02 | check_inbox_driver                   |   Passed        | 00:01:30  |
| 03 | ovt_verify_pkg_install               |   Passed        | 00:06:56  |
| 04 | ovt_verify_status                    |   Passed        | 00:03:21  |
| 05 | vgauth_check_service                 |   Passed        | 00:01:21  |
| 06 | host_verify_saml_token               |   Passed        | 00:02:26  |
| 07 | check_ip_address                     |   Passed        | 00:01:10  |
| 08 | check_os_fullname                    |   Passed        | 00:01:41  |
| 09 | stat_balloon                         |   Passed        | 00:01:05  |
| 10 | stat_hosttime                        |   Passed        | 00:01:11  |
| 11 | device_list                          |   Passed        | 00:02:58  |
| 12 | check_quiesce_snapshot_custom_script |   Passed        | 00:01:35  |
| 13 | memory_hot_add_basic                 |   Passed        | 00:05:56  |
| 14 | cpu_hot_add_basic                    |   Passed        | 00:05:49  |
| 15 | cpu_multicores_per_socket            |   Passed        | 00:08:25  |
| 16 | check_efi_firmware                   |   Passed        | 00:01:19  |
| 17 | secureboot_enable_disable            |   Passed        | 00:04:47  |
| 18 | e1000e_network_device_ops            |   Passed        | 00:04:50  |
| 19 | vmxnet3_network_device_ops           |   Passed        | 00:03:51  |
| 20 | pvrdma_network_device_ops            |   Passed        | 00:08:30  |
| 21 | gosc_perl_dhcp                       |   Passed        | 00:06:14  |
| 22 | gosc_perl_staticip                   |   Passed        | 00:06:00  |
| 23 | gosc_cloudinit_dhcp                  |   Passed        | 00:07:15  |
| 24 | gosc_cloudinit_staticip              |   Passed        | 00:07:15  |
| 25 | paravirtual_vhba_device_ops          |   Passed        | 00:07:32  |
| 26 | lsilogic_vhba_device_ops             | * Not Supported | 00:01:03  |
| 27 | lsilogicsas_vhba_device_ops          | * Not Supported | 00:01:08  |
| 28 | sata_vhba_device_ops                 |   Passed        | 00:07:44  |
| 29 | nvme_vhba_device_ops                 |   Passed        | 00:07:32  |
| 30 | nvdimm_cold_add_remove               |   Passed        | 00:07:30  |
| 31 | ovt_verify_pkg_uninstall             |   Passed        | 00:03:28  |
+-------------------------------------------------------------------------+
```

**Comment by Tomorrow9** [comment]
```
VM information:
+-------------------------------------------------------------------------------------------+
| Name                      | test_vm                                                       |
+-------------------------------------------------------------------------------------------+
| Guest OS Distribution     | Microsoft Windows 11 Enterprise 10.0.26100.0 64-bit           |
+-------------------------------------------------------------------------------------------+
| IP                        |                                              |
+-------------------------------------------------------------------------------------------+
| Hardware Version          |                                                         |
+-------------------------------------------------------------------------------------------+
| VMTools Version           |                                |
+-------------------------------------------------------------------------------------------+
| Config Guest Id           | windows11_64Guest                                             |
+-------------------------------------------------------------------------------------------+
| GuestInfo Guest Id        | windows11_64Guest                                             |
+-------------------------------------------------------------------------------------------+
| GuestInfo Guest Full Name | Microsoft Windows 11 (64-bit)                                 |
+-------------------------------------------------------------------------------------------+
| GuestInfo Guest Family    | windowsGuest                                                  |
+-------------------------------------------------------------------------------------------+
| GuestInfo Detailed Data   | architecture='X86'                                            |
|                           | bitness='64'                                                  |
|                           | buildNumber='26100'                                           |
|                           | distroName='Windows'                                          |
|                           | distroVersion='10.0'                                          |
|                           | familyName='Windows'                                          |
|                           | kernelVersion='26100.1150'                                    |
|                           | prettyName='Windows 11 Enterprise, 64-bit (Build 26100.1150)' |
+-------------------------------------------------------------------------------------------+


Test Results (Total: 30, Passed: 29, Failed: 1, Elapsed Time: 05:00:55)
+---------------------------------------------------------------+
| ID | Name                              |   Status | Exec Time |
+---------------------------------------------------------------+
| 01 | deploy_vm_efi_paravirtual_vmxnet3 |   Passed | 00:48:36  |
| 02 | windows_online_updates_install    |   Passed | 00:49:17  |
| 03 | check_inbox_driver                |   Passed | 00:02:18  |
| 04 | secureboot_enable_disable         |   Passed | 00:09:34  |
| 05 | wintools_complete_install_verify  |   Passed | 00:13:42  |
| 06 | check_efi_firmware                |   Passed | 00:01:04  |
| 07 | check_ip_address                  |   Passed | 00:01:11  |
| 08 | check_os_fullname                 |   Passed | 00:00:59  |
| 09 | mouse_driver_vmtools              |   Passed | 00:00:53  |
| 10 | vgauth_check_service              |   Passed | 00:00:53  |
| 11 | host_verify_saml_token            |   Passed | 00:05:45  |
| 12 | stat_balloon                      |   Passed | 00:00:53  |
| 13 | stat_hosttime                     |   Passed | 00:01:05  |
| 14 | paravirtual_vhba_device_ops       |   Passed | 00:08:26  |
| 15 | lsilogicsas_vhba_device_ops       |   Passed | 00:08:28  |
| 16 | sata_vhba_device_ops              |   Passed | 00:08:26  |
| 17 | nvme_vhba_device_ops              |   Passed | 00:08:31  |
| 18 | nvme_vhba_device_ops_spec13       |   Passed | 00:11:12  |
| 19 | nvme_disk_hot_extend_spec13       |   Passed | 00:07:18  |
| 20 | nvdimm_cold_add_remove            | * Failed | 00:05:00  |
| 21 | e1000e_network_device_ops         |   Passed | 00:07:12  |
| 22 | vmxnet3_network_device_ops        |   Passed | 00:06:00  |
| 23 | memory_hot_add_basic              |   Passed | 00:06:40  |
| 24 | check_quiesce_snapshot            |   Passed | 00:03:48  |
| 25 | cpu_multicores_per_socket         |   Passed | 00:15:57  |
| 26 | gosc_sanity_staticip              |   Passed | 00:18:42  |
| 27 | gosc_sanity_dhcp                  |   Passed | 00:25:39  |
| 28 | vtpm_cold_add_remove              |   Passed | 00:09:11  |
| 29 | wintools_uninstall_verify         |   Passed | 00:07:41  |
| 30 | cpu_hot_add_basic                 |   Passed | 00:04:57  |
+---------------------------------------------------------------+

```

File: windows/eflow_deploy/eflow_deploy.yml [hunk 1]
```diff
@@ -26,7 +26,7 @@
             skip_reason: "Not Supported"
           when:
             - guest_os_edition | lower not in ['pro', 'enterprise']
-            - guest_os_product_type | lower == 'client'
+            - guest_os_product_type == 'client'
```
Comment (keirazhang) [inline windows/eflow_deploy/eflow_deploy.yml:29]: The include_tasks names can also be added.

File: windows/network_device_ops/e1000e_network_device_ops.yml [hunk 2]
```diff
@@ -17,7 +17,7 @@
 
         # Prepare router VM, vSwitch and portgroup
         - include_tasks: ../../common/network_testbed_setup.yml
-          when: router_vm_deployed is undefined or not router_vm_deployed | bool
+          when: router_vm_deployed is undefined or not router_vm_deployed
```
Comment (keirazhang) [inline windows/network_device_ops/e1000e_network_device_ops.yml:20]: The comment can be set as task name

File: windows/secureboot_enable_disable/secureboot_enable_disable.yml [hunk 3]
```diff
@@ -25,11 +25,12 @@
 
         - include_tasks: ../../common/skip_test_case.yml
           vars:
-            skip_msg: "Skip test case due to VM firmware is not EFI: {{ vm_firmware | default('NA') }}"
+            skip_msg: "Skip test case due to VM firmware is not EFI: {{ vm_firmware }}"
```
Comment (keirazhang) [inline windows/secureboot_enable_disable/secureboot_enable_disable.yml:28]: I think message `Skip test case due to VM firmware is {{ vm_firmware }} not EFI` is much better.

File: windows/wintools_uninstall_verify/wintools_uninstall_verify.yml [hunk 4]
```diff
@@ -33,7 +33,7 @@
           when: not vmtools_is_running_before
 
         - include_tasks: prepare_pvscsi_vmxnet3_device.yml
```
Comment (keirazhang) [inline windows/wintools_uninstall_verify/wintools_uninstall_verify.yml:35]: Here can add a task name

File: windows/wintools_uninstall_verify/wintools_uninstall_verify.yml [hunk 5]
```diff
@@ -61,7 +60,7 @@
               vars:
                 vswitch_name: "{{ new_device_vswitch }}"
           when:
-            - guest_os_with_inbox_drivers | bool
+            - guest_os_with_inbox_drivers
```
Comment (keirazhang) [inline windows/wintools_uninstall_verify/wintools_uninstall_verify.yml:63]: This when can be moved before block. And I saw there're some include tasks having names some not. As it is a big amount of tasks names, it's up to you whether to add them in this fix.

File: windows/wsl_distro_install_uninstall/install_wsl.yml [hunk 6]
```diff
@@ -34,7 +34,7 @@
       vars:
         win_powershell_cmd: "{{ wsl_install_cmd }}"
         win_execute_cmd_ignore_error: true
-  when: (guest_os_product_type | lower == 'client')
+  when: (guest_os_product_type == 'client')
```
Comment (keirazhang) [inline windows/wsl_distro_install_uninstall/install_wsl.yml:37]: The `()` can be removed

File: windows/wsl_distro_install_uninstall/install_wsl.yml [hunk 7]
```diff
@@ -60,7 +60,7 @@
       ansible.builtin.set_fact:
         skip_set_wsl_ver: true
       when: guest_os_build_num | int <= 23048
-  when: (guest_os_product_type | lower == 'server')
+  when: (guest_os_product_type == 'server')
```
Comment (keirazhang) [inline windows/wsl_distro_install_uninstall/install_wsl.yml:63]: Same here

File: windows/eflow_deploy/deploy_eflow_vm.yml [hunk 8]
```diff
@@ -83,7 +83,7 @@
       ansible.builtin.set_fact:
         eflow_vm_static_ip: "{{ ('.').join(win_vswitch_ip.split('.')[0:-1]) ~ '.249' }}"
         eflow_vm_gateway_ip: "{{ win_vswitch_gateway_ip }}"
-  when: guest_os_product_type | lower == 'server'
+  when: guest_os_product_type == 'server'
```
[the remark this answers is not in the document]
Comment (Tomorrow9) [reply windows/eflow_deploy/deploy_eflow_vm.yml:86]: Ok, moved, thanks.

File: windows/eflow_deploy/eflow_deploy.yml (hunk 1 above)
Comment (Tomorrow9) [reply windows/eflow_deploy/eflow_deploy.yml:29]: Added.

File: windows/secureboot_enable_disable/secureboot_enable_disable.yml (hunk 3 above)
Comment (Tomorrow9) [reply windows/secureboot_enable_disable/secureboot_enable_disable.yml:28]: Changed, thanks.

File: windows/wsl_distro_install_uninstall/install_wsl.yml (hunk 6 above)
Comment (Tomorrow9) [reply windows/wsl_distro_install_uninstall/install_wsl.yml:37]: Removed, thanks.

File: windows/wintools_uninstall_verify/wintools_uninstall_verify.yml (hunk 5 above)
Comment (Tomorrow9) [reply windows/wintools_uninstall_verify/wintools_uninstall_verify.yml:63]: I'll add task names in the files affected in this merge request firstly. Thanks.

**Review by 123lzxm — COMMENTED** [review]
Have you tested with existing VM? If not, I suggest you test one.

File: windows/utils/win_get_set_disk_online.yml [hunk 9]
```diff
@@ -38,7 +38,7 @@
     win_powershell_cmd: "{{ win_cmd_online_disk }}"
   when:
     - win_online_disk_ops is defined
-    - win_online_disk_ops | lower == 'set'
+    - win_online_disk_ops == 'set'
```
Comment (123lzxm) [inline windows/utils/win_get_set_disk_online.yml:41]: It seems nowhere to set default value for var "win_online_disk_ops" as the parameter's description in line 10.

Comment (Tomorrow9) [reply windows/utils/win_get_set_disk_online.yml]: It's not set explicitly but the logic is the default. Changed to be more clear. Thanks.

**Comment by Tomorrow9** [comment]
> Have you tested with existing VM? If not, I suggest you test one.

Do you mean cover testing on physical testbed or existing VM? What's the concern?

**Comment by 123lzxm** [comment]
> > Have you tested with existing VM? If not, I suggest you test one.
> 
> Do you mean cover testing on physical testbed or existing VM? What's the concern?

Existing VM. To avoid the scenarios that the vars are only initialized in new VM.

File: windows/utils/win_get_set_disk_online.yml [hunk 10]
```diff
@@ -51,11 +51,5 @@
     - win_powershell_cmd_output.stdout_lines | length != 0
     - win_powershell_cmd_output.stdout_lines[0] | lower == 'online'
 
-- name: "Check disk online status"
```
Comment (123lzxm) [inline windows/utils/win_get_set_disk_online.yml:54]: The assert task can be moved after Print task.

File: windows/check_efi_firmware/check_efi_firmware.yml [hunk 11]
```diff
@@ -10,27 +10,30 @@
...
-        - include_tasks: ../setup/test_setup.yml
-        - name: Check VM configured firmware type
+    - name: "Test case block"
+      block:
+        - name: "Test setup"
+          include_tasks: ../setup/test_setup.yml
+        - name: "Get VM configured firmware type"
           include_tasks: ../../common/vm_get_config.yml
           vars:
             property_list: ['config.firmware']
```
[the remark this answers is not in the document]
Comment (Tomorrow9) [reply windows/check_efi_firmware/check_efi_firmware.yml]: Changed, thanks.

File: windows/utils/win_get_set_disk_online.yml (hunk 10 above)
Comment (Tomorrow9) [reply windows/utils/win_get_set_disk_online.yml:54]: Removed 'assert' since here just getting the disk status.

**Review by keirazhang — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**